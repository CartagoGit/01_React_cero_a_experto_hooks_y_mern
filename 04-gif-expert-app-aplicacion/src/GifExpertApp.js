import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({defaultCategories = 'Busca un gif' }) => {
	// const categories = ["One Punch", "Samurai X", "Dragon ball"];
	const [category, setCategories] = useState(defaultCategories);

	// const handleAdd = () =>{
	//     const element = 'Jei';
	//     // setCategories([...categories,element]); //Una solucion
	//     setCategories( c => [...c, element]);
	// };

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory categories={category} setCategories={setCategories} />
			<hr />
			{/* <button onClick={handleAdd}>Agregar</button> */}
			<ol>
				{/* {categories.map((category) => (
					<GifGrid key={category} category={category} />
				))} */}
				<GifGrid key={category} category={category} />
			</ol>
		</>
	);
};
