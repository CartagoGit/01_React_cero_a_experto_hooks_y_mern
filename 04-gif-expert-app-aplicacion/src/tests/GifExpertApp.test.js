import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import {GifExpertApp} from '../GifExpertApp';

describe("Pruebas en GifExpertApp.js", () => {
	let wrapper;

	const fillWrapper = () => {
		wrapper = shallow(<GifExpertApp />);
	};
	beforeEach(() => fillWrapper());

	test("Deberia hacer match con snapshoot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
