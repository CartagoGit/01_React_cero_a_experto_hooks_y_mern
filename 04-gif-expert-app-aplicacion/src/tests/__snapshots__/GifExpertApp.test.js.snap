// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp.js Deberia hacer match con snapshoot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories="Busca un gif"
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Busca un gif"
      key="Busca un gif"
    />
  </ol>
</Fragment>
`;
